lib/graphql/schema/subscription.rb

```diff
@@ -29,7 +29,7 @@ class NoUpdateError < EarlyTerminationError
       # propagate null.
       null false
 
-      def initialize(object:, context:)
+      def initialize(object:, context:, field: nil)
```

Comment:
I was trying to avoid breaking existing implementations where people already have base resolver classes that inherits from `GraphQL::Schema::Resolver` and overwrites its `#initialize` method. Those implementations won't have a `field:` keyword argument, and this PR would break their implementations IMO.

I use a base resolver class like this at work:

```
module Resolvers
  class Base < GraphQL::Schema::Resolver
    include Helpers::Authorization
    
    attr_accessor :current_user, :current_user_role

    InteractError = Helpers::Errors::InteractError

    def initialize(object:, context:)
      @current_user = context.current_user
      @current_user_role = context.current_user&.current_role

      check_authorization!

      super
    end
  end
end
```

But it's up to you whether to make it optional or not since you got more context.

Thanks

cc @rmosolgo

Reply:
Oh, yes, this will be a small breaking change. But I'm not _too_ worried about it because it's easy to fix and it's very easy to find when upgrading. 

Does the `field: nil` in this file actually fix the case above? This class is `Subscription`, the base class for resolvers on the `subscription` root, so it would only affect subscription fields. If you use resolvers for other fields, they'll still be "broken". 

But I'm open to rolling out this change as a "breaking fix" 😅

Reply:
@rmosolgo this make Resolver initalize field arg required 4c1d38be6